src/Proxy.NET/Routes/AzureAISearch.cs

```diff
@@ -0,0 +1,62 @@
+using System.Net;
+using Microsoft.AspNetCore.Mvc;
+using Proxy.NET.Models;
+using Proxy.NET.Services;
+
+namespace Proxy.NET.Routes;
+
+public static class AzureAISearch
+{
+    public static RouteGroupBuilder MapAzureAISearchRoutes(this RouteGroupBuilder builder)
+    {
+        builder.MapPost("/indexes/{index}/docs/search", ProcessRequestAsync).WithMetadata(new Auth(Auth.Type.ApiKey));
+        builder.MapPost("/indexes('{index}')/docs/search.post.search", ProcessRequestAsync).WithMetadata(new Auth(Auth.Type.ApiKey));
+        return builder;
+    }
+
+    private static async Task<IResult> ProcessRequestAsync(
+        [FromServices] ICatalogService catalogService,
+        [FromServices] IProxyService proxyService,
+        [FromServices] IRequestService requestService,
+        HttpContext context,
+        string index
+    )
+    {
+        var routePattern = (context.GetEndpoint() as RouteEndpoint)?.RoutePattern.RawText;
+        var extPath = routePattern?.Split("/indexes").Last();
+        var requestContext = requestService.GetRequestContext() as RequestContext;
+        var apiVersion = ApiVersion(context);
+
+        var requestString = await new StreamReader(context.Request.Body).ReadToEndAsync();
+        if (string.IsNullOrEmpty(requestString))
+            throw new HttpRequestException("Invalid JSON object in body of request", null, HttpStatusCode.BadRequest);
+
+        requestContext!.DeploymentName = index!;
```

Comment:
This clearly suggests there's something in the route that we want to be getting the aspnet router to provide to us since we're having to "work around" nullability.

Reply:
fixed